### ISSUE TITLE
Clusterplans have sensible write rate limits

### ISSUE THREAD

**Issue by lenaschoenburg** [camunda/camunda#20269]
```[tasklist]
### Tasks
- [x] Discover applicable cluster plans
- [x] Decide on relatively high default values for write rate limit and throttling
- [x] Review and apply settings
- [ ] https://github.com/camunda/camunda/issues/20440
```

**Comment by rodrigo-lourenco-lopes** [comment]
Findings from deploying several process loops on clusters of size small, medium and large, in order to find peak processing and exporting rates: 

Small:

- Peak processing rate ~ 6 200
- Peak exporting rate ~ 4 000

Medium: 

- Peak processing rate ~10 000
- Peak exporting rate ~ 7 500

Large: 

-  Peak processing rate ~ 14 000
- Peak exporting rate ~ 7 500

From these values, to keep a balance in the partition's processing and exporting, the write limit values should be set somewhere around the lowest peak rate of the two.

So for the `writeRateLimit.limit` I suggest values around:
S: 4 000
M: 7 500
L: 7 500

For the `writeRateLimit.throttling.minRate` this should probably be somewhat proportional to the previous values with a much smaller base, so following the already used values where the `minRate` is a tenth of the `limit`, I would suggest:
S: 400
M: 750
L: 750

These might need still some further though, as having a min rate of 400 or 750 still might be too high.

For the `writeRateLimit.throttling.acceptableBacklog` since this refers to the queue of not exported records that is stored in disk, it might be useful to use the [disk size of the clusters as guidence](https://docs.camunda.io/docs/components/best-practices/architecture/sizing-your-environment/#sizing-your-runtime-environment):



<img width="970" alt="image" src="https://github.com/user-attachments/assets/78df5e93-3254-42ca-b53e-aa68da46066e">


The acceptable backlog should also be kept below 500 000 as this is the value when alerts get triggered, correct @lenaschoenburg ?

Even though that S and M clusters have the same disk size, the M cluster can clean (export) the backlog faster than the S, so I would maybe recommend:

S: 100 000
M: 200 000
L: 300 000

Wdyt @lenaschoenburg ?

**Comment by lenaschoenburg** [comment]
Thanks for collecting these numbers @rodrigo-lourenco-lopes :+1: 

I'm quite impressed with the throughput (both processing an exporting). Are those per-partition or aggregated?
Could you share links or screenshots for the metrics during your tests? For reference and so that we can dig deeper.

**Comment by rodrigo-lourenco-lopes** [comment]
These were per partition processing and exporting. I did not save screenshots but I can re-do the experience.

**Comment by lenaschoenburg** [comment]
You can also just link to the metrics dashboards, they are still available. No need to redo the experiment.

**Comment by rodrigo-lourenco-lopes** [comment]
[Small cluster ](https://grafana-worker-chaos-1.internal.ultrawombat.com/d/zeebe-dashboard/zeebe?var-namespace=10e6e38b-aff6-4b64-93f1-738c84cf8128-zeebe&var-DS_PROMETHEUS=Thanos&orgId=1&from=1721584415944&to=1721590607800)
[Medium cluster](https://grafana-worker-chaos-1.internal.ultrawombat.com/d/zeebe-dashboard/zeebe?var-namespace=34517e05-5316-41ae-ac92-99b828a92e0a-zeebe&var-DS_PROMETHEUS=thanos&orgId=1&from=1721587264361&to=1721589218122&var-cluster=All&var-pod=All&var-partition=All)
[Large cluster](https://grafana-worker-chaos-1.internal.ultrawombat.com/d/zeebe-dashboard/zeebe?var-namespace=37271a6a-98c9-4252-8f8c-98709898c574-zeebe&var-DS_PROMETHEUS=thanos&orgId=1&from=1721587988188&to=1721591345036&var-cluster=All&var-pod=All&var-partition=All)

**Comment by lenaschoenburg** [comment]
I'd adjust the numbers slightly. We shouldn't look for peaks but for average sustained throughput per partition.

For small:
![small](https://github.com/user-attachments/assets/566fa085-e458-4b80-93a9-ca876921ae98)
Writes per second: ~3k
Exporting per second: ~2k

For medium:
![medium](https://github.com/user-attachments/assets/facdf832-530d-4de6-8d1d-956cd7ca2dea)
Writes per second: ~7k
Exporting per second: ~5k

For large:
![image](https://github.com/user-attachments/assets/6637c2b5-8c81-4c9a-82f9-cfccc7e8306a)
Writes per second: ~11k
Exporting per second: ~7k

**Comment by rodrigo-lourenco-lopes** [comment]
If we use then sustained throughut per partition then we could get the following for the `writeRateLimit.limit`:

S: 2 000
M: 5 000
L: 7 000

And for the `writeRateLimit.throttling.minRate` still using a tenth of the `limit`: 

S: 200
M: 500
L: 700

**Comment by npepinpe** [comment]
Alright, so I think I need to quickly play catch up on the settings and the feature in general in order to better understand.

Generally, if we go by our existing "promises" in SaaS, we say S can do 65 tasks/s - how many records is that, since that's what our limits are based on? Worst case scenario, that's: sequence flow, task activating, task activated, job activate command, job activated, job complete, job completed, task completing, task completed. So that's ~9 records for a task, so ~(65 * 9) or ~585 records/s. Our min rate is below this for S, but our limit is well above. However, my understanding is the min rate is only if our backlog grows too much, or processing slows down too much. So in general we should be above and likely still meet our target. Note that it's unclear to me if this table (from the docs) is promising a minimum, or just giving a general idea. But in general I think the limits look reasonable.

Regarding the `acceptableBacklog`, I don't understand why we pick the numbers based on the max PI living in ES (side note: I don't understand how the table picks those numbers in the first place :sweat_smile:). Can you elaborate?

**Comment by rodrigo-lourenco-lopes** [comment]
From what I understand our minimum rate only happens in case there is quite a catch-up to do, for example, the exporter is severely degraded and barely exporting, if we have the throttling enabled and this has already passed the `aceptableBacklog` then this would process at the minimum rate until our export backlog is smaller than this.

We don't expect these minimum rates to be used unless there is already another existing problem, but since our promissis are significantly higher maybe we could use something as a third of the `limit`, which would give us something like:

S: 600
M: 1500
L: 2100

Would this be in line with our promises in the medium and larger clusters?

> Regarding the acceptableBacklog, I don't understand why we pick the numbers based on the max PI living in ES (side note: I don't understand how the table picks those numbers in the first place 😅). Can you elaborate?

I am not sure if understood the question, the backlog refers to the number of records not exported, which are saved on disk correct? These also can trigger alarms after 500 000 I think. That's why we had settled for a smaller number than that.

**Comment by npepinpe** [comment]
Sounds good. Can you summarize the proposed changes with a short explanation for Felix? Just so he's aware. Then we should open an issue on the operator side to propose those changes, and they can help you walk through testing and deploying them. This would be a great time to have some form of canary deployment (e.g. rollout to 10%), but oh well :smile:

**Comment by rodrigo-lourenco-lopes** [comment]
@lenaschoenburg and I discussed again the minimum write rates and we noticed that if the promise is 65 tasks per second which roughly translates to 585 records per second, then the initially proposed values would make sense if we are talking per cluster, and these values are defined by partition.

So on a small cluster with 3 partitions at a minimum rate of 200 each, we have an effective min rate per cluster of 600 which should be enough :)

**Comment by rodrigo-lourenco-lopes** [comment]
The values in prod were already applied and verified 🚀
